instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py

```diff
@@ -197,6 +197,9 @@ def process_request(self, request):
 
         is_asgi_request = _is_asgi_request(request)
         if not _is_asgi_supported and is_asgi_request:
```

Comment:
Maybe something like this?

```suggestion

        self._unsupported_asgi_warned = False
        ...
        if not _is_asgi_supported and is_asgi_request and not self._unsupported_asgi_warned:
            ...
            self._unsupported_asgi_warned
```

Could be a good idea, or a terrible one :shrug: